python/docs/cli.rst

```diff
@@ -0,0 +1,231 @@
+      -f, --force    force branch assignment
+      --json         write output in json format.
+      -v, --verbose  Verbose output.
+
+      --help         Show this message and exit.
+
+
+
+Remote Command
+--------------
+Set and view the remote. The ``add`` command is a shortcut to ``nessie config --set endpoint <endpoint>`` and the show
+command functions similarly to the ``git remote show <remote>`` command to show the remote and remote refs.
+
+.. code-block:: bash
+
+    Usage: nessie remote [OPTIONS] COMMAND [ARGS]...
+
+      Set and view remote endpoint.
+
+    Options:
+      --help  Show this message and exit.
+
+    Commands:
+      add   Set current remote.
+      show  Show current remote.
+
+
+Log Command
+-----------
+
+View the commit log. This operats similarly to ``git log`` and shows the log in the terminals pager. Revision range is
+specified as <hash>..<hash> or <hash/ref>.
+
+.. code-block:: bash
+
+    Usage: nessie log [OPTIONS] [REVISION_RANGE] [PATHS]...
+
+      Show commit log.
+
+      REVISION_RANGE optional branch, tag or hash to start viewing log from. If of
+      the form <hash>..<hash> only show log for given range\n PATHS optional list
+      of paths. If given, only show commits which affected the given paths
+
+    Options:
+      -n, --number INTEGER    number of log entries to return
+      --since, --after TEXT   Commits more recent than specific date
+      --until, --before TEXT  Commits older than specific date
+      --author, --committer   limit commits to specific committer
+      --json                  write output in json format.
+      --help                  Show this message and exit.
+
+Merge Command
+-------------
+
+Perform a merge operation. This takes commits on ``MERGE_BRANCH`` which not present on ``branch`` and adds them to
+branch.
+
+.. code-block:: bash
+
+    Usage: nessie merge [OPTIONS] [MERGE_BRANCH]
+
+      Merge BRANCH into current branch. BRANCH can be a hash or branch.
+
+    Options:
+      -b, --branch TEXT  branch to cherry-pick onto. If not supplied the default
+                         branch from config is used
+
+      --help             Show this message and exit.
+
+Merge Command
+-------------
+
+Perform a cherry-pick operation. This takes the list of commits ``HASHES`` and adds them to ``branch``.
+
+.. code-block:: bash
+
+    Usage: nessie cherry-pick [OPTIONS] [HASHES]...
+
+      Transplant HASHES onto current branch.
+
+    Options:
+      -b, --branch TEXT  branch to cherry-pick onto. If not supplied the default
+                         branch from config is used
+
+      --help             Show this message and exit.
+
+Contents Command
+----------------
+
+View and list contents.
+
+.. code-block:: bash
+
+    Usage: nessie contents [OPTIONS] [KEY]
+
+      Contents operations.
+
+      KEY name of object to view, delete. If listing the key will limit by
+      namespace what is included.
+
+    Options:
+      -l, --list     list tables
```

Comment:
we should have an option here to pick the ref we want.

Also, how do I create a key like "foo bar"."my friend"?